tests/functional/tests/basic/test_pyocf.py

```diff
@@ -84,3 +84,41 @@ def test_load_cache_recovery(pyocf_ctx):
     cache.stop()
 
     cache = Cache.load_from_device(device_copy)
+
+
+def test_load_cache_with_cores(pyocf_ctx):
+    cache_device = Volume(S.from_MiB(30))
+    core_device = Volume(S.from_MiB(30))
+
+    cache = Cache.start_on_device(cache_device)
+    core = Core.using_device(core_device)
+
+    cache.add_core(core)
+
+    write_data = Data.from_string("This is test data")
+    io = core.new_io(cache.get_default_queue(), S.from_sector(3).B,
+                     write_data.size, IoDir.WRITE, 0, 0)
+    io.set_data(write_data)
+
+    cmpl = OcfCompletion([("err", c_int)])
+    io.callback = cmpl.callback
+    io.submit()
+    cmpl.wait()
+
+    cache.stop()
+
+    cache = Cache.load_from_device(cache_device)
+    cache.add_core(core, try_add=True)
```

Comment:
this is a workaround for our test framework not opening cores inside load? If so, please add a comment explaining this

Reply:
.. or, even better, move this "try_add" inside cache::load_from_device() so that other tests can easily benefit from using it. I think it is worth it even if it requires adding extra param to load_from_device() with a list of cores.

Reply:
I don't think it's a workaround - it's just more explicit way of managing cores while loading, OCL does that too. Right now cores are added automatically only if they're added to core pool - doing that implicitly would be buggy and non-obvious IMO